docs/upgrading.rst

```diff
@@ -0,0 +1,447 @@
+----------------------------------------------
+
+The signature of several methods has changed to have keyword-only arguments.
+For the most part, this should not require any changes, but here's a list of
+the changes.
+
+`parser.error_handler` methods:
+
+.. code-block:: python
+
+    # webargs 5.x
+    def handle_error(error, req, schema, status_code, headers):
+        ...
+
+
+    # webargs 6.x
+    def handle_error(error, req, schema, *, status_code, headers):
+        ...
+
+`parser.__init__` methods:
+
+.. code-block:: python
+
+    # webargs 5.x
+    def __init__(self, location=None, error_handler=None, schema_class=None):
+        ...
+
+
+    # webargs 6.x
+    def __init__(self, location=None, *, error_handler=None, schema_class=None):
+        ...
+
+`parser.parse`, `parser.use_args`, and `parser.use_kwargs` methods:
+
+
+.. code-block:: python
+
+    # webargs 5.x
+    def parse(
+        self,
+        argmap,
+        req=None,
+        location=None,
+        validate=None,
+        error_status_code=None,
+        error_headers=None,
+    ):
+        ...
+
+
+    # webargs 6.x
+    def parse(
+        self,
+        argmap,
+        req=None,
+        *,
+        location=None,
+        validate=None,
+        error_status_code=None,
+        error_headers=None
+    ):
+        ...
+
+
+    # webargs 5.x
+    def use_args(
+        self,
+        argmap,
+        req=None,
+        location=None,
+        as_kwargs=False,
+        validate=None,
+        error_status_code=None,
+        error_headers=None,
+    ):
+        ...
+
+
+    # webargs 6.x
+    def use_args(
+        self,
+        argmap,
+        req=None,
+        *,
+        location=None,
+        as_kwargs=False,
+        validate=None,
+        error_status_code=None,
+        error_headers=None
+    ):
+        ...
+
+
+    # use_kwargs is just an alias for use_args with as_kwargs=True
+
+and finally, the `dict2schema` function:
+
+.. code-block:: python
+
+    # webargs 5.x
+    def dict2schema(dct, schema_class=ma.Schema):
+        ...
+
+
+    # webargs 6.x
+    def dict2schema(dct, *, schema_class=ma.Schema):
+        ...
+
+
+PyramidParser now appends arguments (used to prepend)
+-----------------------------------------------------
+
+`PyramidParser.use_args` was not conformant with the other parsers in webargs.
+While all other parsers added new arguments to the end of the argument list of
+a decorated view function, the Pyramid implementation added them to the front
+of the argument list. This has been corrected, but as a result pyramid views
+with `use_args` will need to be rewritten like so:
+
+.. code-block:: python
+
+    from webargs.pyramidparser import use_args
+
+    # webargs 5.x
+    @use_args({"q": ma.fields.String()}, locations=("query",))
+    def viewfunc(q, request):
+        ...
+
+
+    # webargs 6.x
+    @use_args({"q": ma.fields.String()}, location="query")
+    def viewfunc(request, q):
+        ...
```

Comment:
I didn't double-check but I think request was always first and this change only affects multiple calls to `use_args`.

Reply:
I'll check this and rewrite if I got it wrong.